Fix support for DNS in local-up-cluster.sh

rbtcollins opened the pull request:
[![Analytics](https://kubernetes-site.appspot.com/UA-36037335-10/GitHub/.github/PULL_REQUEST_TEMPLATE.md?pixel)]()

The default appears to be gce now, so this tries to spawn a service in
gce... not the desired outcome for a local test setup.

We also need to sync the kubectl config in the script with the instructions for users, since if people don't test with DNS enabled, it bitrots (which this aims to fix).

To test manually, run something like
```
sudo KUBE_ENABLE_CLUSTER_DNS=true API_HOST_IP=0.0.0.0 ./hack/local-up-cluster.sh
```

The API_HOST_IP=0.0.0.0 is required, otherwise the master is not contactable by kubedns, and the dns pod will fail readychecking on the healthz pod.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/28458)

<!-- Reviewable:end -->

zmerlynn commented:
@k8s-bot ok to test

zmerlynn commented:
@rbtcollins: You're going to have to resolve the CLA issue before we can take this. Thanks!

rbtcollins commented:
CLA should be fixed now (delay was internal)

rbtcollins commented:
@zmerlynn  CLA should be fixed now.

zmerlynn commented:
@rbtcollins: Revisiting this, it looks like you're still having a CLA issue. I see you last attempted this in https://github.com/kubernetes/kubernetes.github.io/pull/948 . It's a corporate CLA?

zmerlynn commented:
There's a way for @googlebot to recognize corporate CLAs as well, but I'm not sure how to set it up on your end (I believe it involves setting up a Google group that you join). Since this is would be your first contribution, can you email me from your VMware corporate account at the email in my profile? I'll look up the corporate CLA on our side while that happens.

rbtcollins commented:
Done; sorry for the delay - was travelling and the email sat in the 'I'll get to that pile' until the bot reminded me.

rbtcollins commented:
ping @zmerlynn

zmerlynn commented:
CLA verified manually by email/corporate verification.

rbtcollins commented:
ping (anyone just needs the merge button clicked now, yes?)

rbtcollins commented:
@zmerlynn ping again.. is this able to be merged at this point ?

Commit:
Fix support for DNS in local-up-cluster.sh

- Fix unbound variables in hack/local-up-cluster.sh

- explicitly set KUBERNETES_PROVIDER in local-up-cluster.
  The default appears to be gce now, so this tries to spawn a service
  in gce... not the desired outcome for a local test setup.

- Remove stale creation of kube-system: the kube-system namespace is
  built in these days.

- Pickup third party etcd automatically.

- Retab hack/local-up-cluster.sh for consistency

- Update hack-local-up's internal credentials to match the user
  instructions we give (which work).

- Note the CA root requirement in local-up-cluster

- Fix a dangling comment in local-up-cluster

To test manually, run something like:
```sudo KUBE_ENABLE_CLUSTER_DNS+true API_HOST_IP=0.0.0.0
   ./hack/local-up-cluster.sh```
The API_HOST_IP=0.0.0.0 is required otherwise kubedns cannot contact
the master and the dns pod will fail readychecking on the healthz
container.

rbtcollins commented:
@zmerlynn sorry to ping you, but I think I got this all hammered out, if you could eyeball it again :)  -thanks!

zmerlynn commented:
@rbtcollins: Sorry for the long delay. Can you squash commits before I LGTM?

rbtcollins commented:
@zmerlynn done!

rbtcollins commented:
@zmerlynn this seems to be stuck - the queue is green AFAICT but the submit queue report says its not.

zmerlynn commented:
@k8s-bot test this
@rbtcollins It looks like this fell victim to one of a couple of instances where we just dropped a bunch of webhooks, not sure.

rbtcollins commented:
@k8s-bot test this
(Can I trigger that? Looks like the GCE test failed all on its own)

rbtcollins commented:
@zmerlynn sorry to bug you - more dropped webhooks I guess?

rmmh commented:
@k8s-bot kubemark test this
